lisdf/parsing/parse_sdf.py

```diff
@@ -1,23 +1,52 @@
 import os
 
-from lisdf.parsing.sdf import SDF
+from lisdf.parsing.sdf import SDF, Collision, Link, Mesh, Visual
 
 
-def load_sdf(model_filename: str):
-    sdf_examples_dir = os.path.join(os.path.dirname(__file__), "../../models")
-    sdf_path = os.path.join(sdf_examples_dir, model_filename)
+def inject_absolute_path(sdf: SDF, model_path: str):
+
+    """
+    This function replace relative paths to object and material
+    files with absolute paths so the sdf object is self-contained.
+    """
+    for wi, world in enumerate(sdf.aggregate_order):
+        for model in world.models:
+            for component in model.aggregate_order:
+                if isinstance(component, Link):
+                    for link_component in component.aggregate_order:
+                        if isinstance(link_component, Collision):
+                            if (
+                                "geometry" in link_component.__dict__.keys()
+                                and isinstance(link_component.geometry, Mesh)
+                            ):
+                                link_component.geometry.uri = os.path.join(
+                                    model_path, link_component.geometry.uri
+                                )
+                        if isinstance(link_component, Visual):
+                            if (
+                                "geometry" in link_component.__dict__.keys()
+                                and isinstance(link_component.geometry, Mesh)
+                            ):
+                                link_component.geometry.uri = os.path.join(
+                                    model_path, link_component.geometry.uri
+                                )
```

Comment:
It looks like the code within the `if` checks do the same thing.

Would `isinstance(link_component, (Collision, Visual))` work here instead?

Reply:
Also are there any other types we would want to inject the URIs? Looking at sdf documentation there is quite a lot of different elements but they may not be relevant for us: http://sdformat.org/spec?elem=link

Reply:
I didn't notice the copied code in the `if` - I agree with Will that it'd be preferable to merge these into a single statement!